extension/src/lib/chromeSettings.js

```diff
@@ -57,22 +70,145 @@ fluid.defaults("gpii.chrome.settings", {
                 flowManager: "ws://localhost:8081/browserChannel",
                 retryTime: 10
             }
+        },
+        contextMenuPanel: {
+            type: "gpii.chrome.settingsContextPanel",
+            options: {
+                model: "{settings}.model",
+                distributeOptions: {
+                    reset: {
+                        target: "{that reset}.options.invokers.click",
+                        record: {
+                            func: "{settings}.updateSettings",
+                            args: ["{settings}.options.defaultSettings"]
+                        }
+                    }
+                }
+            }
         }
-    },
-    model: {
-        settings: "{settings}.options.defaultSettings"
-    },
-    invokers: {
-        updateSettings: {
-            funcName: "gpii.chrome.settings.updateSettings",
-            args: ["{that}",  "{arguments}.0"]
-        }
-    },
-    listeners: {
-        "{wsConnector}.events.onSettingsChange": "{settings}.updateSettings"
     }
 });
 
 gpii.chrome.settings.updateSettings = function (that, settings) {
     that.applier.change("settings", settings || that.options.defaultSettings);
 };
+
+fluid.defaults("gpii.chrome.settingsContextPanel", {
+    gradeNames: ["gpii.chrome.contextMenuPanel"],
+    strings: {
+        captions: "youtube captions",
+        inputsLarger: "enhance inputs",
+        rightClickToSelect: "right-click to select",
+        selfVoicing: "text-to-speech",
+        simplifiedUI: "reading mode",
+        syllabification: "syllables",
+        tableOfContents: "table of contents"
+    },
+    events: {
+        afterContextMenuItemsCreated: null
+    },
+    listeners: {
+        "onCreate.createContextMenuItems": "gpii.chrome.contextMenuPanel.createContextMenuItems"
+    },
+    members: {
+        order: {
+            expander: {
+                funcName: "gpii.chrome.contextMenuPanel.getOrder",
+                args: ["{that}.options.components"]
```

Comment:
Looking at {that}.options.components is also not a portable strategy (to future framework) - you must inspect the components in place once they have constructed.

Reply:
Is there a way to find the child components and to verify they are of the type I'm looking for?

Reply:
https://docs.fluidproject.org/infusion/development/IoCAPI.html#fluidqueryiocselectorroot-selector-flat is a reasonable method